src/passes/DeadCodeElimination.cpp

```diff
@@ -321,84 +322,62 @@ struct DeadCodeElimination : public WalkerPass<PostWalker<DeadCodeElimination>>
     }
   }
 
-  void visitSetLocal(SetLocal* curr) {
-    if (isUnreachable(curr->value)) {
-      replaceCurrent(curr->value);
+  // Append the reachable operands of the current node to a block, and replace
+  // it with the block
+  void BlockifyReachableOperands(std::vector<Expression*> list, WasmType type) {
```

Comment:
Actually I just looked at clang's IR output and it's almost exactly the same all of those ways because it changes the signature anyway.